RewardsSummary/RewardsSummary.WindowsPhone/MainPage.xaml.cs

```diff
@@ -139,6 +219,15 @@ public MainPage()
             }
         }
 
+        private void GetNearbyStores(String storeName)
+        {
+            var webAddr = "https://services-qa.walgreens.com/api/stores/search";
+            var httpWebRequest = (HttpWebRequest)WebRequest.Create(webAddr);
+            httpWebRequest.ContentType = "application/json";
+            httpWebRequest.Method = "POST";
+            httpWebRequest.BeginGetRequestStream(new AsyncCallback(GetRequestStreamCallback), httpWebRequest);
```

Comment:
I guess BeginGetRequestStream() is getting called from the UI thread? Probably not ideal if so, but we can see how it goes.